Source/CarthageKit/Git.swift

```diff
@@ -400,10 +400,15 @@ public func isGitRepository(directoryURL: NSURL) -> SignalProducer<Bool, NoError
 
 	return launchGitTask([ "rev-parse", "--git-dir", ], repositoryFileURL: directoryURL)
 		|> map { outputIncludingLineEndings in
-			let relativeGitDirectory = outputIncludingLineEndings.stringByTrimmingCharactersInSet(.newlineCharacterSet())
-			let gitDirectory = directoryURL.URLByAppendingPathComponent(relativeGitDirectory).path
+			let relativeOrAbsoluteGitDirectory = outputIncludingLineEndings.stringByTrimmingCharactersInSet(.newlineCharacterSet())
+			var absoluteGitDirectory: String
+			if (relativeOrAbsoluteGitDirectory as NSString).absolutePath {
+				absoluteGitDirectory = relativeOrAbsoluteGitDirectory
+			} else {
+				absoluteGitDirectory = (directoryURL.path! as NSString).stringByAppendingPathComponent(relativeOrAbsoluteGitDirectory)
```

Comment:
It would be as follows:
```
var absoluteGitDirectory: String?
if (relativeOrAbsoluteGitDirectory as NSString).absolutePath {
	absoluteGitDirectory = relativeOrAbsoluteGitDirectory
} else {
	absoluteGitDirectory = directoryURL.URLByAppendingPathComponent(relativeOrAbsoluteGitDirectory).path
}
var isDirectory: ObjCBool = false
let directoryExists = absoluteGitDirectory.map { NSFileManager.defaultManager().fileExistsAtPath($0, isDirectory: &isDirectory) } ?? false
```